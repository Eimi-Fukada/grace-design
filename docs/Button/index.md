```jsx
import { Button } from 'grace-design'

export default () => <Button />
```
