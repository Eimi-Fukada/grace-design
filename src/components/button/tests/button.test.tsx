import '@testing-library/jest-dom'
import { fireEvent, render, screen } from '@testing-library/react'
import React from 'react'
import Button from '../'

const classPrefix = `gd-button`

test('should match snapshot', () => {
  const { container } = render(<Button color="primary">主要按钮</Button>)
  expect(container.firstChild?.nodeName).toBe('BUTTON')
  expect(container).toMatchSnapshot()
})

test('should children correctly', () => {
  const { getByText, getByTestId } = render(
    <Button
      data-testid="button"
      className="aa"
      style={{ margin: 8 }}
      color="primary"
    >
      主要按钮
    </Button>
  )
  expect(getByText('主要按钮')).toHaveTextContent('主要按钮')
  expect(getByTestId('button')).toHaveClass('aa')
  expect(getByTestId('button')).toHaveAttribute('style')
})

test('renders with color', () => {
  const { getByText } = render(
    <>
      <Button color="primary">Primary</Button>
      <Button color="success">Success</Button>
      <Button color="danger">Danger</Button>
      <Button color="warning">Warning</Button>
    </>
  )
  expect(getByText('Primary')).toHaveClass(`${classPrefix}-primary`)
  expect(getByText('Success')).toHaveClass(`${classPrefix}-success`)
  expect(getByText('Danger')).toHaveClass(`${classPrefix}-danger`)
  expect(getByText('Warning')).toHaveClass(`${classPrefix}-warning`)
})

test('renders with block', () => {
  const { getByText } = render(
    <Button block color="primary">
      Block
    </Button>
  )
  expect(getByText('Block')).toHaveClass(`${classPrefix}-block`)
})

test('renders with disabled', () => {
  const { getByRole } = render(<Button disabled>Disabled</Button>)
  const button = getByRole('button')
  expect(button).toBeDisabled()
})

test('renders with loading and loadingText', () => {
  const { getByTestId } = render(
    <Button loading loadingText="加载中" data-testid="btn">
      Loading
    </Button>
  )
  expect(getByTestId('btn')).toHaveClass(`${classPrefix}-loading`)

  // children no render
  expect(() => {
    screen.getByText('Loading')
  }).toThrow(/Unable to find an element with the text: Loading./)

  screen.getByText('加载中')
})

test('renders with type', () => {
  const { getByRole } = render(<Button type="submit">submit</Button>)
  const button = getByRole('button')
  expect(button).toHaveAttribute('type', 'submit')
  expect(button).not.toHaveAttribute('type', 'button')
})

test('renders with onClick', () => {
  class DefaultButton extends React.Component {
    state = {
      loading: false,
    }

    enterLoading = () => {
      this.setState({ loading: true })
    }

    render() {
      const { loading } = this.state
      return (
        <Button
          loading={loading}
          loadingText="加载中"
          onClick={this.enterLoading}
        >
          Button
        </Button>
      )
    }
  }
  const { getByText } = render(<DefaultButton />)
  fireEvent.click(getByText('Button'))
  screen.getByText('加载中')
})
