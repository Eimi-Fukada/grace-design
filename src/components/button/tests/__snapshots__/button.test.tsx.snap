// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <button
    class="gd-button gd-button-primary"
    type="button"
  >
    主要按钮
  </button>
</div>
`;
